### PR TITLE
28 - Update to Polymer 2.x - part 3

### PR DESCRIPTION
**Pull request by dotpointer**
Update to Polymer 2.x - part 3.

* Removing unnecessary constructor.
* Replacing `fire` with `dispatchEvent` in tests.
* Removing `is` attribute from style in `custom-style`.

Not sure if the dom-repeat template should be wrapped in `dom-repeat`.

I am also welcoming suggestions on more things to convert in this component, if there are any.

Issue is #28.

### CODE REVIEW

**Comment by Wurper** [comment]
`this.async` can be replaced with `Polymer.Async.microTask.run(() => this.someMethod());`
And import `<link rel="import" href="/bower_components/polymer/lib/utils/async.html">`

File: test/basic.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
 						// Set input value
 						autocomplete.text = value.toString();
 						// Trigger create suggestions
-						suggestions._input.fire('keyup');
+						suggestions.dispatchEvent(new CustomEvent('keyup', { bubbles: true, composed: true }));
```
Comment (Wurper) [inline test/basic.html]: Is there a reason for removing `_input` ?

File: test/basic.html [hunk 2]
```diff
@@ -118,7 +118,7 @@
 						done();
 					}, 200);
 
-					suggestions._input.fire('focus', {});
+					suggestions.dispatchEvent(new CustomEvent('focus', { bubbles: true, composed: true }));
```
Comment (Wurper) [inline test/basic.html]: Same as above

File: test/basic.html (hunk 1 above)
Comment (dotpointer) [reply test/basic.html]: No, this was a mistake that also halted the tests. Added it back.

File: test/basic.html (hunk 2 above)
Comment (dotpointer) [reply test/basic.html]: No, this was a mistake that also halted the tests. Added it back.

**Comment by dotpointer** [comment]
@Wurper Do you know if `run`has the duration in the second parameter?

`Polymer.Async.microTask.run(() => this.someMethod() /*, <timeout here?*/);`

It seemed to pass the tests, but not sure if it really passed the timeout.

I cannot check because the documentation refuses to load.

fail

**Comment by Wurper** [comment]
@dotpointer 
According to https://polymer-library.polymer-project.org/2.0/docs/upgrade
you should use native setTimeout if a delay is used.

I checked the polymer source (if I looked at the correct one) and it looks like it puts in a microtask queue. No consideration to any delay.
https://github.com/Polymer/polymer/blob/665901ab2483385cf6ae7037663ad0f992e9b930/lib/utils/async.js

**Comment by dotpointer** [comment]
@Wurper Replacing it with `window.setTimeout` then, pushed the change.

File: paper-autocomplete-chips.html [hunk 3]
```diff
@@ -389,9 +387,8 @@
 				return;
 			}
 			this.push('selectedItems', newSelection);
-			this.async(function () {
+			Polymer.Async.microTask.run(function () {
```
Comment (Wurper) [inline paper-autocomplete-chips.html]: Maybe change to arrow function and skip bind(this)?

**Pull request merged**